pennylane/tape/qscript.py

```diff
@@ -383,6 +383,7 @@ def _update(self):
             # Invalidate cached properties so they get recalculated
             del self.wires
             del self.par_info
+            del self.hash
```

Comment:
Would be good to update the `test_cached_properties` test to include the hash.
Do we need additional tests to verify that caching the hash works and does not equate unequal tapes in some way?

Reply:
thanks for the reminder. turns out we weren't refreshing `trainable_params` :(

Reply:
Thanks for fixing it!